src/modules/configuration/src/lib/Configuration.c

```diff
@@ -0,0 +1,547 @@
+        {
+            snprintf(buffer, sizeof(buffer), "%u", g_refreshInterval);
+        }
+        else if (0 == strcmp(objectName, g_localManagementEnabledObject))
+        {
+            snprintf(buffer, sizeof(buffer), "%s", g_localManagementEnabled ? "true" : "false");
+        }
+        else if (0 == strcmp(objectName, g_fullLoggingEnabledObject))
+        {
+            snprintf(buffer, sizeof(buffer), "%s", g_fullLoggingEnabled ? "true" : "false");
+        }
+        else if (0 == strcmp(objectName, g_commandLoggingEnabledObject))
+        {
+            snprintf(buffer, sizeof(buffer), "%s", g_commandLoggingEnabled ? "true" : "false");
+        }
+        else if (0 == strcmp(objectName, g_iotHubProtocolObject))
+        {
+            snprintf(buffer, sizeof(buffer), "%u", g_iotHubProtocol);
+        }
+        else
+        {
+            OsConfigLogError(ConfigurationGetLog(), "MmiGet called for an unsupported object (%s)", objectName);
+            status = EINVAL;
+        }
+    }
+
+    if (MMI_OK == status)
+    {
+        *payloadSizeBytes = strlen(buffer);
+
+        if ((g_maxPayloadSizeBytes > 0) && ((unsigned)*payloadSizeBytes > g_maxPayloadSizeBytes))
+        {
+            OsConfigLogError(ConfigurationGetLog(), "MmiGet(%s, %s) insufficient maxmimum size (%d bytes) versus data size (%d bytes), reported buffer will be truncated",
+                componentName, objectName, g_maxPayloadSizeBytes, *payloadSizeBytes);
+
+            *payloadSizeBytes = g_maxPayloadSizeBytes;
+        }
+
+        *payload = (MMI_JSON_STRING)malloc(*payloadSizeBytes);
+        if (*payload)
+        {
+            memcpy(*payload, buffer, *payloadSizeBytes);
+        }
+        else
+        {
+            OsConfigLogError(ConfigurationGetLog(), "MmiGet: failed to allocate %d bytes", *payloadSizeBytes + 1);
+            *payloadSizeBytes = 0;
+            status = ENOMEM;
+        }
+    }    
+
+    if (IsFullLoggingEnabled())
+    {
+        OsConfigLogInfo(ConfigurationGetLog(), "MmiGet(%p, %s, %s, %.*s, %d) returning %d", clientSession, componentName, objectName, *payloadSizeBytes, *payload, *payloadSizeBytes, status);
+    }
+
+    FREE_MEMORY(configuration);
+
+    return status;
+}
+
+int ConfigurationMmiSet(MMI_HANDLE clientSession, const char* componentName, const char* objectName, const MMI_JSON_STRING payload, const int payloadSizeBytes)
+{
+    int status = MMI_OK;
+
+    char* payloadString = NULL;
+
+    int modelVersion = g_modelVersion;
+    int refreshInterval = g_refreshInterval;
+    int localManagementEnabled = g_localManagementEnabled ? 1 : 0;
+    int fullLoggingEnabled = g_fullLoggingEnabled ? 1 : 0;
+    int commandLoggingEnabled = g_commandLoggingEnabled ? 1 : 0;
+    int iotHubProtocol = g_iotHubProtocol;
+
+    JSON_Value* jsonValue = NULL;
+    JSON_Object* jsonObject = NULL;
+
+    if ((NULL == componentName) || (NULL == objectName) || (NULL == payload) || (0 >= payloadSizeBytes))

```

Comment:
payloadSizeBytes = 0 should be valid (no limit)

Reply:
MmiSet with payloadSizeBytes (aka no payload) is not valid. This is not the parameter for MmiOpen, this is MmiSet.